platform-api/che-core-api-git/src/main/java/org/eclipse/che/api/git/GitService.java

```diff
@@ -391,6 +388,17 @@ public void tagDelete(TagDeleteRequest request) throws ApiException {
                     String[] keyValues = row.split("=", 2);
                     result.put(keyValues[0], keyValues[1]);
                 }
+            } else if (request.isSet()) {
+                for (Map.Entry<String, String> configData : request.getConfigData().entrySet()) {
```

Comment:
Please rework current method to a GET api method with query parameter like this:
`@GET
Map<String, String> getConfigs(@QueryParam List<String> requestedConfigs) {
//returns all if requestedConfigs.isEmpty()
...
}`

and add a new `setConfig` PUT api method with `ConfigRequest` dto like this: 
`@PUT
void setConfig(ConfigRequest config) {
...
}`.

The ConfigRequest DTO should contain only information map of config entries to update, like this:
`@DTO
public interface ConfigRequest {
    Map<String, String> getConfigEntries();
    void setConfigEntries(Map<String, String> configEntries);
    ConfigRequest withConfigEntries(Map<String, String> configEntries);
}`